### PR TITLE
Support html5 links inside Shadow DOM

### PR DESCRIPTION
**Pull request by fraxu**
We are using web components so we needed to get html5 links inside shadow dom. So we needed to get correct target element from MouseEvent's composedPath() instead of just using the target property. 

We also added a test that clicks links to test this. Test will fail with "Some of your tests did a full page reload!" if test fails.

If you want any changes to this, I'm happy to help!

### CODE REVIEW

File: modules/reitit-frontend/src/reitit/frontend/history.cljs [hunk 1]
```diff
@@ -59,6 +59,14 @@
           el
           (recur (.-parentNode el)))))))
 
+(defn- event-target [event]
+  ;; Read event's target from composed path to get shadow dom working,
+  ;; fallback to target property if not available
```
Comment (alexanderjamesking) [inline modules/reitit-frontend/src/reitit/frontend/history.cljs]: This would be better in a doc-string instead of a comment

Comment (fraxu) [reply modules/reitit-frontend/src/reitit/frontend/history.cljs]: Thanks for the input. I changed this to doc-string.

File: modules/reitit-frontend/src/reitit/frontend/history.cljs [hunk 2]
```diff
@@ -59,6 +59,14 @@
           el
           (recur (.-parentNode el)))))))
 
+(defn- event-target [event]
+  "Read event's target from composed path to get shadow dom working,
+   fallback to target property if not available"
+  (let [original-event (.-event_ event)]
```
Comment (Deraen) [inline modules/reitit-frontend/src/reitit/frontend/history.cljs]: Use `.getBrowserEvent` method instead of the internal `event_` property.

**Comment by Deraen** [comment]
I will need to read a bit to understand this better, I don't really have any experience with Shadow DOM. Luckily the change is quite small, I'll try to find time to check this properly tomorrow.

File: modules/reitit-frontend/src/reitit/frontend/history.cljs (hunk 2 above)
Comment (fraxu) [reply modules/reitit-frontend/src/reitit/frontend/history.cljs]: Changed

**Pull request merged**